.github/workflows/ci.yml

```diff
@@ -0,0 +1,445 @@
+                --node-version=20 \
+                --platform=darwin \
+                --arch=arm64
+
+              for f in ./builds/*-darwin-universal*; do "$f"; done
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '20'
+
+      - uses: actions/download-artifact@v2
+        with:
+          name: builds
+          path: ./builds
+
+      - name: Build
+        run: ${{ matrix.script }}
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: builds
+          path: ./builds
+
+  integration-prerelease:
+    name: "Integration / Pre-release"
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/matrixai/github-runner
+
+    concurrency:
+      group: integration-prerelease
+      cancel-in-progress: true
+
+    needs:
+      - build-prerelease
+      - integration-platforms
+      - integration-deployment
+
+    if: github.ref == 'refs/heads/staging'
+
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          name: builds
+          path: ./builds
+
+      - name: Publish pre-release
+        env:
+          CI_REGISTRY_USER: ${{ github.actor }}
+          CI_REGISTRY_PASSWORD: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          echo 'Publishing application prerelease'
+
+          nix develop .#ci --command bash -c $'
+            if gh release view "$CI_COMMIT_TAG" --repo "$GH_PROJECT_PATH" >/dev/null; then \
+              gh release \
+                upload "$GITHUB_REF_NAME" \
+                builds/*.closure.gz \
+                builds/*-docker-* \
+                builds/*-linux-* \
+                builds/*-win32-* \
+                builds/*-darwin-* \
+                --clobber \
+                --repo "$GH_PROJECT_PATH"; \
+            else \
+              gh release \
+                create "$CI_COMMIT_TAG" \
+                builds/*.closure.gz \
+                builds/*-docker-* \
+                builds/*-linux-* \
+                builds/*-win32-* \
+                builds/*-darwin-* \
+                --title "${GITHUB_REF_NAME}-$(date -u +"%Y-%m-%dT%H:%M:%SZ")" \
+                --notes "" \
+                --prerelease \
+                --target staging \
+                --repo "$GH_PROJECT_PATH"; \
+            fi;
+          '
+
+          echo 'Prereleasing container image'
+
+          nix develop .#ci --command bash -c $'
+            skopeo login \
+              --username "$CI_REGISTRY_USER" \
+              --password "$CI_REGISTRY_PASSWORD" \
+              "$CI_REGISTRY_IMAGE";
+            image=(./builds/*-docker-*);
+            ./scripts/deploy-image.sh "${image[0]}" \'testnet\' "$CI_REGISTRY_IMAGE";
+          '
+
+  integration-merge:
+    name: "Integration / Merge"
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/matrixai/github-runner
+
+    concurrency:
+      group: integration-merge
+      cancel-in-progress: true
```

Comment:
Same here. If a new commit is pushed up to staging, why are we cancelling the merge?